src/components/PhotoDetails/PhotoDetails.js

```diff
@@ -1,27 +1,52 @@
 import React, { Component, PropTypes as T } from 'react';
 import { Link } from 'react-router';
 import { connect } from 'react-redux';
+import axios from 'axios';
 
-import { getPhotoDetails } from '../../actions/index';
+import { getPhotoDetails, getComments } from '../../actions/index';
 import RelatedPhotoGallery from '../RelatedPhotoGallery/RelatedPhotoGallery';
 import Header from '../Header/Header';
+import CommentForm from '../Forms/CommentForm';
 
 class PhotoDetails extends Component {
   componentDidMount() {
     this.props.getPhotoDetails(this.props.params.id);
+    this.props.getComments(this.props.params.id);
   }
+
+  componentWillReceiveProps(propUpdate) {
+    if (this.props.params.id !== propUpdate.params.id) {
+      this.props.getPhotoDetails(propUpdate.params.id);
+    }
+  }
+
   render() {
-    const { photo, comments } = this.props;
+    const { photo, comments, commentForm } = this.props;
+    const handlePostComment = e => {
+      e.preventDefault();
+      axios.post(`https://500pxserver-zuuynfmrvy.now.sh/api/photos/${this.props.params.id}/comments`, {
+        text: commentForm.values.userComment,
+        username: commentForm.values.username,
+      })
+      .then(() => {
+        this.props.getComments(this.props.params.id);
+      })
+      .catch(error => {
+        throw error;
+      });
+    };
     return (
       <div>
         <Header/>
         <div className="container">
+          <p><Link to="/">Back</Link></p>
           <h2>{photo.name}</h2>
           <img className="z-depth-4" src={photo.image_url}/>
           <h5>Comments</h5>
```

Comment:
Purely optional organizational thing – maybe we want to move any current comments so they show up above the form (form appears at the bottom of any comments puled in), and the votes_count is moved out of the comments section?